src/Frontend/Core/Engine/Url.php

```diff
@@ -364,9 +364,12 @@ private function processQueryString()
 
             // remove language
             if ($hasMultiLanguages) {
-                $URL = trim(str_replace('/' . $language, '', $URL), '/');
+                $URL = str_replace('/' . $language, '', $URL);
             }
 
+            // remove the first slash
+            $URL = trim($URL, '/');
```

Comment:
It would be better but with the original it was:
`$URL = trim(str_replace('/' . $language, '', $URL), '/')`